### ISSUE TITLE
 add systemd socket activation support to mpd through exit on idle mode

### ISSUE THREAD

**Issue by twojstaryzdomu** [MusicPlayerDaemon/MPD#1317]
## Feature request
This is a request to add systemd socket activation support to mpd. This may be useful for better use of resources when mpd isn't playing anything and no client is connected.
Mpd may be started with systemd's socket activation mechanism when a client connects to the mpd port. The chief benefit is quitting on idle. Specifically, when no song is played within the idle threshold, mpd will exit. The socket in systemd will return to the listening state as the mpd process has exit with the zero status.
That way, neither the mpd service nor the mpd process have to be started ahead of the user running the client. Only the systemd socket needs to be running, which is more efficient resource-wise.
To enable socket activation, the systemd socket for mpd needs to set the ReusePort flag to allow co-existence with the mpd process. Mpd needs to support the TCP SO_REUSEPORT flag internally as well. Clients require no change.
When coupled with the proposed idle time out parameter, mpd will quit after some arbitrary amount of time has elapsed when idle. A client becoming active will start mpd again when a user specifically needs it. Playing music would reset the idle interval.

Please note, this isn't an inetd-style activation, as using one normally spawns a new process for each connection.

I have been using a similar socket activation mechanism for other applications. It requires minor internal changes (TCP port flag) but is quite beneficial in low memory cases (RPi 0). Think of it as a way to run mpd on demand, without it running constantly once started.

The support for this mode could be activated through the optional exit on idle parameter, that would take an arbitrary number of seconds to quit after, and also enable TCP ReusePort flag, necessary for socket activation.

When in the mode, running a scan would stop the idle timer until it is done. Starting playback would stop the timer as well. Stopping playback would start the idle timer. When the idle timer elapses, the mpd process would exit with the zero status. That way, mpd quitting after idle will not be flagged as a failure by systemd and consequently not be suppressing further socket activations.

**Comment by MaxKellermann** [comment]
TLDR, because your issue started with something that already exists. Since the code is open, you can do all the checks yourself, and save me the time for doing that work for you.
